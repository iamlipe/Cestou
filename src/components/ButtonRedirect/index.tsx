import React, {memo} from 'react';
import styled, {useTheme} from 'styled-components/native';
import Icon from 'react-native-vector-icons/MaterialIcons';

interface Props {
  title: string;
  onPress: () => void;
}

const ButtonRedirect = ({title, onPress}: Props) => {
  const theme = useTheme();

  return (
    <StyledContainer onPress={onPress}>
      <StyledText>{title}</StyledText>
      <Icon
        name="keyboard-arrow-right"
        color={theme.colors.GRAY_900}
        size={24}
      />
    </StyledContainer>
  );
};

const StyledContainer = styled.TouchableOpacity`
  width: 100%;
  height: 49px;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: ${({theme}) => theme.colors.PRIMARY_100};
  border-radius: 4px;
  border: 1px solid ${({theme}) => theme.colors.PRIMARY_300};
  padding: 0 16px;
  margin-bottom: 16px;
`;

const StyledText = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
`;

export default memo(ButtonRedirect);
