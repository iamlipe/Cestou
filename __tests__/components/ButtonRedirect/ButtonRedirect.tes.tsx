import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/Testing';

import ButtonRedirect from '@/components/ButtonRedirect';

describe('ButtonRedirect', () => {
  const onPress = jest.fn();
  const titleText = 'press me';

  test('should render correctly', () => {
    renderWithThemeProvider(
      <ButtonRedirect title={titleText} onPress={onPress} />,
    );
  });

  test('shoudl matches snapshot', () => {
    matchSnapshotWithProvider(
      <ButtonRedirect title={titleText} onPress={onPress} />,
    );
  });

  test('should render title', () => {
    const buttonRedirect = renderWithThemeProvider(
      <ButtonRedirect title={titleText} onPress={onPress} />,
    );

    const title = buttonRedirect.getByText(titleText);

    expect(title).toBeTruthy();
  });
});
