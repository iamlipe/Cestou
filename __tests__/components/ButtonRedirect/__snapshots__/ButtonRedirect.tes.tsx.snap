// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRedirect shoudl matches snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#C1E9CF",
      "borderColor": "#6CCD91",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 49,
      "justifyContent": "space-between",
      "marginBottom": 16,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 0,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#1A1A1A",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 27,
        },
      ]
    }
  >
    press me
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#1A1A1A",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    îŒ•
  </Text>
</View>
`;
